packages/legend-application-repl/src/stores/dataCube/grid/DataCubeGridState.ts

```diff
@@ -139,4 +148,166 @@ export class DataCubeGridState extends DataCubeQuerySnapshotSubscriber {
+        );
+      };
+      reader.readAsDataURL(blob);
+    });
+  };
+
+  generateEmail = async (isHtml: boolean, includeAttachments: boolean) => {
+    try {
+      const textContent =
+        'Please find the attached CSV and Excel(.xlsx) files.';
+      const htmlContent = `
+      <html>
+        <body>
+          <p>${textContent}</p>
+        <body>
+      </html>
+      `;
+
+      const emailBody = isHtml ? htmlContent : textContent;
+      const contentType = isHtml ? 'text/html' : 'text/plain';
+      const boundaryMixed = 'boundary----------123456789----------987654321';
+      const boundaryAlternative =
+        'boundary----------0987654321----------1234567890';
+
+      let emlContent = `
+From: somebody@test.com
+To: someone@test.com
+Subject: subject of the some files generated
+MIME-Version: 1.0
+Content-Type: multipart/mixed; boundary="${boundaryMixed}"
+
+--${boundaryMixed}
+Content-Type: multipart/alternative; boundary="${boundaryAlternative}"
+
+--${boundaryAlternative}
+Content-Type: text/plain; charset="UTF-8"
+Content-Transfer-Encoding: 7bit
+
+${textContent}
+
+--${boundaryAlternative}
+Content-Type: text/html; charset="UTF-8"
+Content-Transfer-Encoding: 7bit
+
+${htmlContent}
+
+--${boundaryAlternative}--
+
+`;
+
+      if (includeAttachments) {
+        const csvContent =
+          this._client?.getDataAsCsv({
+            fileName: 'csv_data.csv',
+            columnSeparator: ',',
+            suppressQuotes: true,
+          }) ?? '';
+        const csvBlob = new Blob([csvContent], { type: 'text/csv' });
+
+        const xlsxContent = this._client?.getDataAsExcel({
+          fileName: 'excel_data.xlsx',
+          processCellCallback: (cell: any) => {
+            return cell ? cell.toString() : '';
+          },
+        });
+        let xlsxBlob;
+        if (xlsxContent instanceof Blob) {
+          xlsxBlob = xlsxContent;
+        } else if (typeof xlsxContent === 'string') {
+          xlsxBlob = new Blob([xlsxContent], {
+            type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+          });
+        } else {
+          throw new Error('Failed to generate Excel content.');
+        }
+
+        emlContent += `
+--${boundaryMixed}
+Content-Type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet; name="excel_data.xlsx"
+Content-Transfer-Encoding: base64
+Content-Disposition: attachment; filename="excel_data.xlsx"
+
+${await this.blobToBase64(xlsxBlob)}
+
+--${boundaryMixed}
+Content-Type: text/csv; name="csv_data.csv"
+Content-Transfer-Encoding: base64
+Content-Disposition: attachment; filename="csv_data.csv"
+
+${await this.blobToBase64(csvBlob)};
+
+--${boundaryMixed}--
+`;
+      } else {
+        emlContent += `
+Content-Type: ${contentType}; charset= "UTF-8"
+Content-Transfer-Encoding: 7bit
+
+${emailBody}
+`;
+      }
+
+      downloadFileUsingDataURI('emlFile.eml', emlContent, ContentType.ALL); //NEED TO ADD MESSAGE_RFC822 to CONTENT-TYPE
```

Comment:
did you finally figure out what we need to pick here? message_rfc822 or */* ?

Reply:
It depends, if we would like to have more flexible with ALL or email only type of the content. I kept it as All. All is like any I would say. what we loose I believe is the precision on a MIME type. Personally me I would go and add message_rfc822. since we might have an issue with the email clients. some clients prefer to avoid ambigiouse MIME types as ALL. should I go ahead and ad that?